crates/gen/src/required_interface.rs

```diff
@@ -9,6 +9,26 @@ pub struct RequiredInterface {
     pub kind: InterfaceKind,
 }
 
+impl PartialEq for RequiredInterface {
+    fn eq(&self, other: &Self) -> bool {
+        self.name == other.name
+    }
+}
+
+impl Eq for RequiredInterface {}
+
+impl PartialOrd for RequiredInterface {
```

Comment:
You should be able to derive all but `PartialEq` no?

Reply:
The `derive` macro is too aggressive. Even though the custom `PartialEq` implementation is provided and deriving `Eq` should be simple, it attempts to validate that all fields are `Eq` when they don't need to be.